bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/type/ChannelType.java

```diff
@@ -41,59 +42,35 @@
     private final ChannelKind kind;
     private final Set<String> tags;
     private final String category;
-    private final StateDescription state;
-    private final CommandDescription commandDescription;
+    private final @Nullable StateDescription state;
+    private final @Nullable CommandDescription commandDescription;
     private final EventDescription event;
     private final URI configDescriptionURI;
     private final AutoUpdatePolicy autoUpdatePolicy;
 
     /**
+     * Creates a new instance of this class with the specified parameters.
+     *
      * @deprecated Use the {@link ChannelTypeBuilder} instead.
      */
     @Deprecated
     public ChannelType(ChannelTypeUID uid, boolean advanced, String itemType, String label, String description,
-            String category, Set<String> tags, StateDescription state, URI configDescriptionURI) {
-        this(uid, advanced, itemType, ChannelKind.STATE, label, description, category, tags, state, null,
-                configDescriptionURI);
+            String category, Set<String> tags, EventDescription event, URI configDescriptionURI)
+            throws IllegalArgumentException {
+        this(uid, advanced, itemType, ChannelKind.TRIGGER, label, description, category, tags, null, null, event,
+                configDescriptionURI, null);
     }
 
     /**
+     * Creates a new instance of this class with the specified parameters.
+     *
      * @deprecated Use the {@link ChannelTypeBuilder} instead.
      */
     @Deprecated
-    public ChannelType(ChannelTypeUID uid, boolean advanced, String itemType, ChannelKind kind, String label,
```

Comment:
Looks like there are still add-ons using these deprecated constructors. I made  local add-ons port build and it failed on the Homematic add-on:

```
HomematicTypeGeneratorImpl.java:[306,14266] The constructor ChannelType(ChannelTypeUID, boolean, String, ChannelKind, String, String, String, null, StateDescription, EventDescription, URI) is undefined
````

So if we'd merge this PR it would break the add-ons port. Unless the deprecations are fixed on the 2.5.x. branch.

Reply:
I am currently working on a PR for the Homematic binding that will replace all deprecated constructors, methods etc. Will try to push it today.

Reply:
That will certainly help @MHerbst! :+1:

Reply:
@MHerbst Thank you. I will crawl the rest of the add-ons and replace the remaining.

@wborn How can I manage a local add-ons port for testing?

Reply:
I use this [port-addons.sh](https://gist.github.com/wborn/0e3cd275be724e827ef88bb04b7eb12a) script based on the script used by Jenkins in the add-ons port build.

Reply:
So you run the script on a 2.5.x based branch.

If you want to do the same build as Jenkins you checkout 2.5.x in that dir. E.g. to see if there are any openhab-core changes breaking the add-ons port build.

Sometimes you also want to checkout your own 2.5.x branch to see if you can keep 2.5.x compatible with breaking changes OH 3.0. I do that for instance when adding nullness annotations. You might also want to do that after fixing deprecations before submitting the PR.

Reply:
My PR to fix add-ons build: https://github.com/openhab/openhab-addons/pull/7354/

Reply:
Thank you both for fixing the deprecations!
The add-ons port now again builds for me with this PR locally. :+1: